pkg/reconciler/build/build_test.go

```diff
@@ -552,5 +552,26 @@ var _ = Describe("Reconcile Build", func() {
 				Expect(err).ToNot(BeNil())
 			})
 		})
+
+		Context("when build object does not have sources, but output timestamp is set to be the source timestamp", func() {
```

Comment:
I see this as two good different test cases. Can you pls keep one just for sources being empty, I think this is missing at the moment.

Then have the one for this PR, just an invalid value for the spec.output.timestamp.

Reply:
The source being empty test case is in E2E: https://github.com/shipwright-io/build/pull/1509/files#diff-0fd53c65ae74ef65351d15ec5b81bc056b2ce9ef35a0211acf2696a9b746b8a5R213-R241

Reply:
And there is another empty source one in the unit tests: https://github.com/shipwright-io/build/pull/1509/files#diff-85436457242b63f7f72d7c9fa4af832b829172d0652898e209eb7b80d4690c2eR557-R574.

Reply:
@qu1queee I am not fully sure what test is missing based on your comment. Is it about one for spec.output.timestamp having an invalid value?

Reply:
I think the first one is already covered as you mentioned. Yes, can we have one test case for an invalid value in the timestamp, so that the Build validations fail?